modules/extension/app-schema/app-schema/src/test/java/org/geotools/http/HttpStaticServer.java

```diff
@@ -0,0 +1,187 @@
+                new AbstractHandler() {
+                    @Override
+                    public void handle(
+                            String target,
+                            Request baseRequest,
+                            HttpServletRequest request,
+                            HttpServletResponse response) {
+                        for (Map.Entry<String, String> resource : resources.entrySet()) {
+                            if (target != null
+                                    && target.equalsIgnoreCase("/" + resource.getKey())) {
+                                // we found the resource we where looking for
+                                response.setContentType("text/xml");
+                                response.setStatus(HttpServletResponse.SC_OK);
+                                baseRequest.setHandled(true);
+                                try (InputStream input =
+                                        new ByteArrayInputStream(resource.getValue().getBytes())) {
+                                    // write the resource content to the HTTP response output stream
+                                    IOUtils.copy(input, response.getOutputStream());
+                                    // we are done
+                                    return;
+                                } catch (Exception exception) {
+                                    throw new RuntimeException(
+                                            "Error writing HTTP response.", exception);
+                                }
+                            }
+                        }
+                        // request not handled
+                        response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+                        baseRequest.setHandled(true);
+                    }
+                });
+    }
+
+    /**
+     * Adds a static resource that should be published by the static HTTP server. The resource will
+     * be read from the classpath and it will be available at: http://{host}:{port}/{resourceName}
+     * Placeholders {host} and {port} will be replace with this server values.
+     */
+    void putResource(String resourceName, String resourcePath) {
+        try (InputStream input = this.getClass().getResourceAsStream(resourcePath)) {
+            if (input == null) {
+                // we could not find the resource int he classpath
+                throw new RuntimeException(
+                        String.format(
+                                "Resource '%s' with path '%s' not found.",
+                                resourceName, resourcePath));
+            }
+            // read the resource from the classpath
+            String resource = IOUtils.toString(input);
+            // substitute host and port place holders
+            String relativePath =
+                    URLs.getParentUrl(this.getClass().getResource(resourcePath)).getPath();
+            resource = resource.replaceAll("\\{relative\\}", relativePath);
```

Comment:
@nmco these parameters should be documented in the user manual. They might also conflict with [app-schema property interpolation](http://docs.geoserver.org/latest/en/user/data/app-schema/property-interpolation.html#using-properties). For example, a user might want to use `${host}` and `${port}`. Consider using the app-schema `InterpolationProperties` class, the `${propertyname}` syntax, and longer property names like `${app-schema.http-server.relative}`, `${app-schema.http-server.host}`, and `${app-schema.http-server.port}`, or even some leading underscores to protect users from surprises. I am not known for brevity so will defer to your good taste and judgement.  :-P

Reply:
@bencaradocdavies this are just placeholders for the tests since the port is randomized :-P